File: benchmarks/correctness/python/vcf_filter.py
"""Filter VCF by QUAL>=30, count variants per chromosome. Output JSON."""
import json
import sys

chrom_counts = {}
total = 0
passed = 0
with open("data/variants.vcf") as f:
    for line in f:
        if line.startswith("#"):
            continue
        total += 1
        fields = line.strip().split("\t")
        qual = float(fields[5]) if fields[5] != "." else 0.0
        if qual >= 30:
            passed += 1
            chrom = fields[0]
            chrom_counts[chrom] = chrom_counts.get(chrom, 0) + 1

# Sort by chromosome name
sorted_counts = dict(sorted(chrom_counts.items()))
json.dump({"total_variants": total, "passed_qual30": passed,
           "per_chromosome": sorted_counts}, sys.stdout, indent=2)
print()

File: benchmarks/correctness/r/vcf_filter.R
# Filter VCF by QUAL>=30, count variants per chromosome. Output JSON.
library(jsonlite)

lines <- readLines("data/variants.vcf")
lines <- lines[!grepl("^#", lines)]
total <- length(lines)

passed_chroms <- c()
for (line in lines) {
    fields <- strsplit(line, "\t")[[1]]
    qual <- suppressWarnings(as.numeric(fields[6]))
    if (!is.na(qual) && qual >= 30) {
        passed_chroms <- c(passed_chroms, fields[1])
    }
}

chrom_counts <- as.list(sort(table(passed_chroms)))
# Convert to integer
chrom_counts <- lapply(chrom_counts, as.integer)

result <- list(
    total_variants = total,
    passed_qual30 = length(passed_chroms),
    per_chromosome = chrom_counts
)
cat(toJSON(result, auto_unbox = TRUE, pretty = TRUE))
cat("\n")
